### ISSUE TITLE
Very slow netrw operations on headless server

### ISSUE THREAD

**Issue by rr-** [neovim/neovim#6048]
- `nvim --version`: `NVIM v0.2.0-498-g050819277`
- Vim (version: ) behaves differently? not sure
- Operating system/version: Arch Linux
- Terminal name/version: cygwin + MinTTY + ssh with x11 forwarding (configured so in `~/.ssh/config`) + tmux (works both within x server and outside of it)
- `$TERM`: `screen-256color`

### Actual behaviour

It takes about 8 seconds for `:Explore` to open netrw window.

### Expected behaviour

It should work instantly, the directory contains about 5 files.

### Steps to reproduce using `nvim -u NORC`

```
time nvim -u NORC -c 'Explore|q'
nvim -u NORC -c 'Explore|q'  0.06s user 0.01s system 0% cpu 9.102 total
```

Running it through `strace` reveals that it spends a great deal of time looking for stuff such as `xclip` and whatnot. `:set clipboard` shows `clipboard=`. It works fast only after I removed the `xclip` binary which I've installed mainly for clipboard via X11 forwarding.

**Comment by justinmk** [comment]
Does it happen every time, or only the first `:Explore`?

netrw loves to fiddle with the system clipboard for no reason (in a _loop_).  `:set clipboard=` doesn't help because netrw messes with `@+` explicitly.

This is arguably a netrw bug. We've already worked around the loop issue. Perhaps  `provider/clipboard.vim` could put a `jobwait()` around its xclip attempt so that it times out after 1s.

**Comment by justinmk** [comment]
Also is X11 forwarding functioning when this happens? Is `$DISPLAY` valid? Does `:let @='foo'` on the remote nvim alter your clipboard?

**Comment by rr-** [comment]
It's slow also on directory traversal but not that much.

Turns out it's slow only when `DISPLAY` is not empty. After manually fixing `export DISPLAY=` nvim is fast again. I'm not sure how `DISPLAY` could have been corrupted but it could have to do with tmux attaching.

**Comment by justinmk** [comment]
Ok, thanks. What does `:CheckHealth provider` say about the clipboard?

**Comment by rr-** [comment]
```
health#provider#check
========================================================================
## Clipboard
  - SUCCESS: Clipboard tool found: xclip

## Python 2 provider
  - WARNING: No Python interpreter was found with the neovim module.  Using the first available for diagnostics.
  - WARNING: provider/pythonx: Could not load Python 2:
    /usr/bin/python2 does not have the neovim module installed. See ":help provider-python".
    /usr/bin/python2.7 does not have the neovim module installed. See ":help provider-python".
    python2.6 not found in search path or not executable.
    /usr/bin/python is Python 3.6 and cannot provide Python 2.
  - ERROR: Python provider error
    - SUGGESTIONS:
      - provider/pythonx: Could not load Python 2:
          /usr/bin/python2 does not have the neovim module installed. See ":help provider-python".
          /usr/bin/python2.7 does not have the neovim module installed. See ":help provider-python".
          python2.6 not found in search path or not executable.
          /usr/bin/python is Python 3.6 and cannot provide Python 2.
  - INFO: Executable: Not found

## Python 3 provider
  - WARNING: No Python interpreter was found with the neovim module.  Using the first available for diagnostics.
  - WARNING: provider/pythonx: Could not load Python 3:
    /usr/bin/python3 does not have the neovim module installed. See ":help provider-python".
    python3.7 not found in search path or not executable.
    /usr/bin/python3.6 does not have the neovim module installed. See ":help provider-python".
    python3.5 not found in search path or not executable.
    python3.4 not found in search path or not executable.
    python3.3 not found in search path or not executable.
    /usr/bin/python does not have the neovim module installed. See ":help provider-python".
  - ERROR: Python provider error
    - SUGGESTIONS:
      - provider/pythonx: Could not load Python 3:
          /usr/bin/python3 does not have the neovim module installed. See ":help provider-python".
          python3.7 not found in search path or not executable.
          /usr/bin/python3.6 does not have the neovim module installed. See ":help provider-python".
          python3.5 not found in search path or not executable.
          python3.4 not found in search path or not executable.
          python3.3 not found in search path or not executable.
          /usr/bin/python does not have the neovim module installed. See ":help provider-python".
  - INFO: Executable: Not found

## Ruby provider
  - INFO: Ruby: ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-linux]
  - WARNING: Missing "neovim" gem.
    - SUGGESTIONS:
      - Run in shell: gem install neovim
```

**Comment by nimitbhardwaj** [comment]
Hey, check again, its fast now
```text
time nvim -u NORC -c 'Explore|q'

real	0m0.324s
user	0m0.168s
sys	0m0.048s
```
These are my stats, not a bad time, even a good time for an application like nvim

**Comment by justinmk** [comment]
Closing in favor of https://github.com/neovim/neovim/issues/11089
